src/google/protobuf/compiler/csharp/csharp_options.h

```diff
@@ -41,14 +41,29 @@ namespace csharp {
 
 // Generator options (used by csharp_generator.cc):
 struct Options {
-  Options() : file_extension(".cs"), base_namespace(""), generate_directories(false) {
+  Options() :
+      file_extension(".cs"),
+      base_namespace(""),
+      base_namespace_specified(false) {
   }
   // Extension of the generated file. Defaults to ".cs"
   string file_extension;
-  // Base namespace to use to create directory hierarchy. Defaults to ""
+  // Base namespace to use to create directory hierarchy. Defaults to "".
+  // This option allows the simple creation of a conventional C# file layout,
+  // where directories are created relative to a project-specific base
+  // namespace. For example, in a project with a base namespace of PetShop, a
+  // proto of user.proto with a C# namespace of PetShop.Model.Shared would
+  // generate Model/Shared/User.cs underneath the specified --csharp_out
+  // directory.
+  //
+  // If no base namespace is specified, all files are generated in the
+  // --csharp_out directory, with no subdirectories created automatically.
```

Comment:
> That's true, but is also the behaviour for js.

It's considered a bug -- there is an internal bug to address this.

If you try to generate both files in a single invocation, protoc will fail and bail rather than actually overwrite one file with another. But if you compile two proto files with separate invocations, it will overwrite one with the other without a warning or error. :(

I'm not a huge fan of this behavior but I'll leave it to you to decide what is best for C# users. :)